src/adservice/Dockerfile

```diff
@@ -12,26 +12,26 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM eclipse-temurin:17-jdk AS builder
+FROM gradle:7-jdk17 AS builder
```

Comment:
I don't recommend using a the gradle wrapper for local builds / development and using a different version of gradle for the docker build.

Instead, you can speed up the docker build by ordering the commands such that things that change infrequently like the gradle wrapper and the dependencies are cached in intermediate layers:
```
FROM eclipse-temurin:17-jdk AS builder

WORKDIR /usr/src/app/

COPY ./src/adservice/gradlew* ./src/adservice/settings.gradle* ./src/adservice/build.gradle .
COPY ./src/adservice/gradle ./gradle

RUN ./gradlew
RUN ./gradlew downloadRepos

COPY ./src/adservice/ ./
COPY ./pb/ ./proto
RUN ./gradlew installDist -PprotoSourceDir=./proto

# -----------------------------------------------------------------------------

FROM eclipse-temurin:17-jre
```

Reply:
Makes sense, let me fix this here (and with the frauddetectionservice, where I did the same)

Reply:
Updated it accordingly